src/js/components/channels/twilio-channel-content.jsx

```diff
@@ -26,8 +26,7 @@ export class TwilioChannelContentComponent extends Component {
 
     handleInputChange = (telNumber) => {
         updateTwilioAttributes({
-            appUserNumber: telNumber,
-            hasError: false
```

Comment:
I reset the `hasError` flag in `onNumberValid` instead. 

When updating the UI following the `link:cancelled` event, `handleInputChange` would get triggered and would reset the `hasError` flag, thus never rendering the message. 😐